penman/tree.py

```diff
@@ -40,6 +40,16 @@ def __str__(self) -> str:
         s = _format(self.node, 2)
         return f'Tree(\n  {s})'
 
+    def __contains__(self, item: str):
+        """
+        Test whether a given string is part of the tree as a terminal instance node.
+        """
+        if isinstance(item, str):
+            terminals = [target for _, (role, target) in self.walk() if role == "/" and is_atomic(target)]
+            return item in terminals
+        else:
+            raise NotImplemented
```

Comment:
It seems like you could look for branches as you did for triples above. Otherwise, same comments as before:
- make a set
- Try not to raise a specific error, or if it's an unsupported type use `TypeError`